rust/src/schema.rs

```diff
@@ -100,6 +100,44 @@ impl Field {
         }
     }
 
+    /// Return Arrow Data Type name.
+    pub fn type_str(t: &DataType) -> String {
+        match t {
+            DataType::Boolean => "bool",
+            DataType::UInt8 => "uint8",
+            DataType::Int8 => "int8",
+            DataType::UInt16 => "uint16",
+            DataType::Int16 => "int16",
+            DataType::UInt32 => "uint32",
+            DataType::Int32 => "int32",
+            DataType::UInt64 => "uint64",
+            DataType::Int64 => "int64",
+            DataType::Float16 => "halffloat",
+            DataType::Float32 => "float",
+            DataType::Float64 => "double",
+            DataType::Binary => "binary",
+            DataType::Utf8 => "string",
+            _ => panic!(),
+        }.to_string()
+    }
+
+    pub fn is_numeric(t: &DataType) -> bool {
+        use DataType::*;
+        matches!(
+            t,
+            UInt8
```

Comment:
those indents are not right?

Aren't `arrow2` has similar check?

Reply:
Actually it's copied from [arrow-rs](https://github.com/apache/arrow-rs/blob/f60841cf9e35b6f7d7098e9c25926cddd725c2d4/arrow/src/datatypes/datatype.rs#L1395) and keeps the original indents, at least Clion believe it indents properly.

Arrow2 also has a [similar one](https://github.com/jorgecarleitao/arrow2/blob/0a1714c18faa1a4c5f10482b5dde97ceea5cad9c/src/compute/cast/mod.rs#L44), but it's not public.

Reply:
Should we not put this as public interface on `Schema` as well?

Reply:
Good idea, private it.